### DISCUSSION TITLE
Extracting Link to Zotero Item

### DISCUSSION THREAD

**Discussion by stefanopagliari** [retorquere/zotero-better-bibtex#1848] [Q&A] [answered]
Is it possible to amend the postscript so that the link to the item on zotero (e.g. zotero://select/items/1_MHJDXTEH) is also exported?
I have tried this, but unsuccessfully:
`reference.add({ name: 'localLibrary', value: item.localLibrary });`

I realize this is not what BBT is designed for, but I'm using the automatic export file to as the starting point to generate literature notes, from where I would like to access the item on zotero.

**Comment by retorquere** [comment]
Not a problem. You can see what's available to the postscript by exporting a sample item using the `BetterBibTeX JSON` format; in your case

```
reference.add({ name: 'localLibrary', value: item.select });
```

would do the job.

**Reply by stefanopagliari** [reply]
This is not working for me. Here is what I have at the moment in the postscript.
```
if (Translator.BetterTeX) {
  reference.add({ name: 'keywords', value: item.tags, sep: ',  ', enc: 'tags' });
  reference.add({ name: 'dateadded', value: item.dateAdded });
  reference.add({ name: 'datemodified', value: item.dateModified });
  reference.add({ name: 'localLibrary', value: item.select });  
  reference.add({ name: 'pdfAttachments', value: item.pdfAttachments });
  reference.add({ name: 'collections', value: item.collections.map(key => Translator.collections[key] && Translator.collections[key].name).filter(name => name) })

}
```

Am I missing something?

**Comment by retorquere** [comment, marked as answer]
My bad -- it should have been

```
const [ , kind, lib, key ] = item.uri.match(/^https?:\/\/zotero\.org\/(users|groups)\/((?:local\/)?[^/]+)\/items\/(.+)/);
reference.add({ name: 'localLibrary', value: (kind === 'users') ? `zotero://select/library/items/${key}` : `zotero://select/groups/${lib}/items/${key}` });
```

I had precalculated `item.select` elsewhere but that isn't available in postscripts.

`item.pdfAttachments` doesn't exist, but if you want the filenames of pdf attachments, you can do

```
reference.add({ name: 'pdfAttachments', value: item.attachments.filter(att => att.mimetype == 'application/pdf').map(att => att.localPath), sep: ', ' })
```

**Reply by stefanopagliari** [reply]
That worked. Thank you again for your help!
